DEVELOPMENT.md

```diff
@@ -300,6 +300,17 @@ These are the general steps to do when replacing old Helm chart(s) with a Module
     +  - crossplane-system/crossplane-packages
     ```
 
+### Testing Modules
+
+Modules should be tested just as components have been tested previously in the [end-to-end tests](tests/end-to-end).
```

Comment:
```suggestion
Modules should be tested just as components have been tested previously in the [tests](tests/).
```
Is there a reason to scope this to end-to-end only right now?
I mean the goal is to move as much things as possible out of here, but for now this would be more "true" to how we do it, and don't limit us to only write expensive tests.